lib/fish.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Fish do
+
+  def next_timer(0), do: {6, 8}
+  def next_timer(n), do: {n - 1, 0}
+
+  def day_tick(fish, 0), do: fish
+  def day_tick(fish, days) do
+    IO.inspect(days)
+    fish
+      |> Enum.reduce({[], []}, fn n, {oldFish, newFish} ->
+        case next_timer(n) do
+          {n, 0} -> {oldFish ++ [n], newFish}
+          {6, 8} -> {oldFish ++ [6], newFish ++ [8]}
+        end
```

Comment:
А вот это уже поторопились. `next_timer/1` умный (почти) и уже отдает правильное сзачение, чтобы избежать `case/2` (почти).

```elixir
def next_timer(0), do: {[6], [8]}
def next_timer(n), do: {[n - 1], []}
```

И тут:

```suggestion
        {old, new} = next_timer(n)
        {oldFish ++ old, newFish + new}
```

Reply:
тоже думал в эту сторону, чуть-чуть не догадался